spec.md

```diff
@@ -673,10 +681,10 @@ message CreateVolumeRequest {
   // topological accessibility information supported by the SP.
   // This field is OPTIONAL.
   // This field SHALL NOT be specified unless the SP has the
-  // ACCESSIBILITY_CONSTRAINTS plugin capability.
+  // VOLUME_ACCESSIBILITY_CONSTRAINTS plugin capability.
   // If this field is not specified and the SP has the
-  // ACCESSIBILITY_CONSTRAINTS plugin capability, the SP MAY choose
-  // where the provisioned volume is accessible from.
+  // VOLUME_ACCESSIBILITY_CONSTRAINTS plugin capability, the SP MAY
+  // choose where the provisioned volume is accessible from.
   TopologyRequirement accessibility_requirements = 7;
```

Comment:
Since a volume MAY be created from a snapshot, perhaps with specific workload placement in mind, it seems that the interesting/valid set of topologies (for on-demand workload placement) MAY be the intersection of the topologies of the (a) candidate nodes; (b) snapshots, and; (c) volume accessibility requirements specified here. Is this actually the case, or am I over-thinking this?

If my thinking is in line with the spirit of what this PR is trying to accomplish, then it would be great to see a bit more description fleshed out in the spec somewhere about the interaction between these three object types and their topologies.

Reply:
In other words, simply adding "or snapshot" to the existing set of documentation (as has been done throughout the PR here) probably isn't sufficient.

Reply:
When creating a volume from snapshot, SP will look at accessibility_requirements in CreateVolumeRequest and accessible_topology in CreateSnapshotResponse, and find an intersection of the two.  This will be reflected in accessible_topology of CreateVolumeResponse.

Regarding accessibility for a node, the existing comment about accessible_topology in NodeGetInfoResponse says “COs MAY use this information along with the topology information returned in CreateVolumeResponse to ensure that a given volume is accessible from a given node when scheduling workloads.”  This comment is still true for creating volume from snapshot.

I can add this to the spec.

Reply:
Spec updated with this clarification.